rust/lance/src/dataset/schema_evolution.rs

```diff
@@ -289,6 +295,24 @@ pub(super) async fn add_columns_to_fragments(
     Ok((fragments, schema))
 }
 
+fn has_legacy_files(fragments: &[FileFragment]) -> bool {
+    let has_legacy_files = fragments
+        .iter()
+        .map(|f| &f.metadata)
+        .flat_map(|fragment_meta| fragment_meta.files.iter())
+        .any(|file_meta| {
+            matches!(
+                LanceFileVersion::try_from_major_minor(
+                    file_meta.file_major_version,
+                    file_meta.file_minor_version
+                ),
+                Ok(LanceFileVersion::Legacy)
+            )
+        });
+
+    !has_legacy_files
+}
```

Comment:
Might be easier to just use `dataset.is_legacy_storage()`.

Reply:
Thanks. I wasn't aware of that method